src/yz_events.erl

```diff
@@ -59,7 +70,26 @@
 %%%===================================================================
 
 start_link() ->
-    gen_server:start_link({local, ?MODULE}, ?MODULE, [], []).
+    gen_event:start_link({local, ?MODULE}).
+
+add_handler(Handler, Args) ->
+    gen_event:add_handler(?MODULE, Handler, Args).
+
+add_sup_handler(Handler, Args) ->
+    gen_event:add_sup_handler(?MODULE, Handler, Args).
+
+add_guarded_handler(Handler, Args) ->
+    yz_eventhandler_sup:start_guarded_handler(?MODULE, Handler, Args).
```

Comment:
I was wondering if we could actually just use riak_core_eventhandler_sup and riak_core_eventhandler, instead of duplicating them in this code.

As I understand the rationale for this guard pattern: We want a process that will effectively act as a kind of monitor for yz_events, such that when the guard terminates (say, as a normal shut down of yokozuna), it will terminate the event handler.  Similarly, if the event handler is "deleted" (i.e. removed, if I understand the event_handler docs), the guard will get a en_event_EXIT message.